### PR TITLE
Feature: add command-line options to override `external-controller`, `secret` and `external-ui` (#531)

### PR DESCRIPTION
**Pull request by ysc3839**
Discussion: #531

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -18,18 +18,30 @@ import (
...
 	flag.StringVar(&homeDir, "d", "", "set configuration directory")
 	flag.StringVar(&configFile, "f", "", "specify configuration file")
+	flag.StringVar(&externalUI, "ext-ui", "", "override external ui directory")
+	flag.StringVar(&externalController, "ext-ctl", "", "override external controller address")
+	flag.StringVar(&secret, "secret", "", "override secret for RESTful API")
 	flag.BoolVar(&version, "v", false, "show current version of clash")
 	flag.BoolVar(&testConfig, "t", false, "test configuration and exit")
 	flag.Parse()
+
+	flagset = make(map[string]bool)
```
Comment (Dreamacro) [inline main.go:41]: maybe we don't need this

Comment (ysc3839) [reply main.go:41]: So without this how to check if a flag exist?

Comment (Dreamacro) [reply main.go:41]: In this case, a not empty string is enough

Comment (ysc3839) [reply main.go:41]: But empty string can still be used to override config. For example: force an empty secret.

Comment (Dreamacro) [reply main.go:41]: what about `flag.Lookup`?

Comment (ysc3839) [reply main.go:41]: Not working, `flag.Lookup("ext-ui")` always returns not nil value.
```
fmt.Printf("ext-ui %t %t\n", flag.Lookup("ext-ui") != nil, flagset["ext-ui"])
```
```
$ ./clash
ext-ui true false
```
```
$ ./clash -ext-ui test
ext-ui true true
```

Comment (Dreamacro) [reply main.go:41]: ok, I didn't expect this library so sucks. recommend to use `map[string]bool{}` and format the code with `go fmt`

Comment (ysc3839) [reply main.go]: Updated.

**Comment by Dreamacro** [comment]
👍Thanks for your contribution

**Pull request merged**